lib/surface/component.ex

```diff
@@ -41,7 +41,7 @@ defmodule Surface.Component do
 
       use Surface.BaseComponent, type: unquote(__MODULE__)
 
-      use Surface.API, include: [:prop, :slot, :data]
+      use Surface.API, include: [:prop, :slot, :data, :upload]
```

Comment:
Not sure if the `upload` is relevant for a stateless component. I do not see a real use case where it is useful. If no one is against it, I suggest removing it. We can always add it if someone has a concrete example.

Reply:
Do you know if LiveView allows `allow_upload` to be used in components (stateful or stateless) or just in the live views? Would suggest keeping it as close as possible to that.

Reply:
allow_upload works in stateful components, using it in a stateless should(would?) just bubble up to the stateful parent imo.

Reply:
@olivermt is right. The component does not care about the @uploads assigns. What is important is the event that are send back to the component or the live view.